__tests__/endpoints.test.js

```diff
@@ -68,15 +71,163 @@ describe("app", () => {
           .get("/api/reviews/this-is-not-an-id")
           .expect(400)
           .then(({ body: { msg } }) => {
-            expect(msg).toBe("Invalid User Id");
+            expect(msg).toBe("Invalid Review Id");
           });
       });
-      test("status 404, responds with error message when passed an id thta does not exist", () => {
+      test("status 404, responds with error message when passed an id that does not exist", () => {
         return request(app)
           .get("/api/reviews/999")
           .expect(404)
           .then(({ body: { msg } }) => {
-            expect(msg).toBe("User Id Not Found");
+            expect(msg).toBe("Review Id Not Found");
+          });
+      });
+    });
+  });
+  describe("PATCH /api/reviews/:review_id", () => {
+    test("status 200, responds with the updated review object, for update that increments votes", () => {
+      const reviewUpdate = { inc_votes: 1 };
+      return request(app)
+        .patch("/api/reviews/1")
+        .expect(200)
+        .send(reviewUpdate)
+        .then(({ body: { review } }) => {
+          expect(review).toBeInstanceOf(Object);
+          expect(review).toEqual(
+            expect.objectContaining({
+              votes: 2,
+            })
+          );
+        });
+    });
+    test("status 200, responds with the updated review object, for update that decrements votes", () => {
+      const reviewUpdate = { inc_votes: -3 };
+      return request(app)
+        .patch("/api/reviews/4")
+        .expect(200)
+        .send(reviewUpdate)
+        .then(({ body: { review } }) => {
+          expect(review).toBeInstanceOf(Object);
+          expect(review).toEqual(
+            expect.objectContaining({
+              votes: 4,
+            })
+          );
+        });
+    });
+    describe("ERRORS", () => {
+      test("status 400, responds with error message if req body is an empty object", () => {
+        const reviewUpdate = {};
+        return request(app)
+          .patch("/api/reviews/2")
+          .expect(400)
+          .send(reviewUpdate)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+      });
+      test("status 400, responds with error message if update is of incorrect type", () => {
+        const reviewUpdate = { inc_votes: "i am a string, not a number!" };
+        return request(app)
+          .patch("/api/reviews/2")
+          .expect(400)
+          .send(reviewUpdate)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+      });
+    });
+  });
+  describe("GET /api/users", () => {
+    test("status 200, responds with an array of user objects", () => {
+      return request(app)
+        .get("/api/users")
+        .expect(200)
+        .then(({ body: { users } }) => {
+          expect(users).toBeInstanceOf(Array);
+          users.forEach((user) => {
+            expect(user).toEqual(
```

Comment:
An empty array will pass this test. Consider adding a length check to make sure you don't get false positives